app/services/courses_query.rb

```diff
@@ -5,10 +5,11 @@ def self.call(params = {})
     new(params:).call
   end
 
-  attr_reader :scope
+  attr_reader :scope, :params
 
   def initialize(params:)
     @params = params
+    @applied_filters = []
```

Comment:
Is there such thing as "unapplied filters"?
If not, I'd just call this `filters`

Reply:
Also, are we mixing domains here?
Filter sounds like a business domain term and queries are data layer.

This feels akin to using view logic in models

Reply:
I could called applied scopes? Or applied parameters.

The only reason I am using to be more quick during development to know which things this query is searching - because the DB query might become too big with 16 filters from the view to understand.

Reply:
I renamed to `applied_scopes`.